### PR TITLE
Initial interface to sale Pix mail

### PR DESCRIPTION
**Pull request by victorabarros**

### CODE REVIEW

File: user-interface/pages/confirmation.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+const Confirmation = () => {
+  return (
+    <div className={'container'}>
+      <title>Meu Pix Fácil</title>
+
+      <main className={'main'}>
+        <p className={'description'}>
+          Agora só resta fazer um pix de R$80 para blablabla@meupixfacil.com
+          <br />
+          e enviar o comprovante para o mesmo email.
```
Comment (fernando-ringba) [inline user-interface/pages/confirmation.js:12]: No need to send the receipt, it is possible to add a note to the pix transfer, clients can use this note to identify themselves.

File: user-interface/components/Form.js [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+  const [errors, setErrors] = useState({});
+  const [message, setMessage] = useState('');
+
+  const domains = [
+    // 'meupix.vip',
+    'meupixfacil.com.br',
+    // 'meupixfacil.com',
+    // 'pixfacil.vip',
+    // 'pixpro.club',
+    'victorbarros.com.br', // TODO remove
```
Comment (fernando-ringba) [inline user-interface/components/Form.js:20]: lets use only meupixfacil.com initially

File: user-interface/components/Form.js [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+
+  const createPixmail = async (form) => {
+    let accResp;
+    try {
+      accResp = await api.createAccount(form);
+      if (![CREATED, OK].includes(accResp.status)) {
+        throw new Error(accResp.data);
+      }
+    } catch (error) {
+      console.log(error);
```
Comment (fernando-ringba) [inline user-interface/components/Form.js:40]: we should log errors in a file, no?

File: user-interface/pages/confirmation.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+const Confirmation = () => {
+  return (
+    <div className={'container'}>
+      <title>Meu Pix Fácil</title>
+
+      <main className={'main'}>
+        <p className={'description'}>
+          Agora só resta fazer um pix de R$80 para blablabla@meupixfacil.com com
+          sua identificação na descrição.
```
Comment (fernando-ringba) [inline user-interface/pages/confirmation.js:11]:
```suggestion
          Agora só resta fazer um PIX de R$79 para meupixfacil@meupixfacil.com adicionando seu email na descrição do PIX.
```

File: user-interface/components/Form.js [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+  };
+
+  const formValidate = () => {
+    let err = {};
+    Object.keys(form).map((prop) => {
+      if (!form[prop]) err[prop] = 'Campo obrigatório';
+    });
+    // TODO usar regex p validar email
+    return err;
+  };
```
Comment (fernando-ringba) [inline user-interface/components/Form.js:94]: There are components that can be used to take care of the validation.

File: user-interface/service/api.js [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+const createPixmail = async (accountId, payload) => {
+  const baseUrl = getBaseUrl();
+  return axiosPost(`${baseUrl}/accounts/${accountId}/pixemail`, payload);
+};
+
+const axiosPost = async (url, payload) => {
+  try {
+    return axios.post(url, payload);
+  } catch (error) {
+    console.log(error);
```
Comment (fernando-ringba) [inline user-interface/service/api.js:28]: we should log errors in a file

**Review by fernando-ringba — CHANGES_REQUESTED** [review]
Check comments.

File: user-interface/components/Form.js (hunk 3 above)
Comment (victorabarros) [reply user-interface/components/Form.js:40]: It will be solve at #12

File: user-interface/components/Form.js (hunk 5 above)
Comment (victorabarros) [reply user-interface/components/Form.js:94]: We need to improve with better tools then regex.
https://react-hook-form.com/
https://medium.com/@adostes/validating-a-form-in-react-cc29d47e140f

**Review by fernando-ringba — APPROVED** [review]

**Pull request merged**